Source/Csla/Configuration/Fluent/CslaOptions.cs

```diff
@@ -31,6 +31,19 @@ public CslaOptions(IServiceCollection services)
       DataPortalClientOptions = new DataPortalClientOptions(this);
     }
 
+    /// <summary>
+    /// Registers a specific ApplicationContext manager service,
+    /// overriding the default service registration.
+    /// </summary>
+    /// <typeparam name="T">IContextManager implementation type.</typeparam>
+    /// <returns></returns>
+    public CslaOptions RegisterContextManager<T>() 
+      where T : IContextManager
+    {
+      Services.AddScoped(typeof(IContextManager), typeof(T));
+      return this;
```

Comment:
Yeah, this is great. Very discoverable from a framework user's perspective.

Long term, we should maybe consider what should be an extension method in a separate file, and what should be within the options class itself. 

There is an option to keep the options classes as simple as possible by having only properties on them. All of the methods could be extension methods. I think this is maybe what is more normal in this type of code. However, this new addition is entirely consistent with the existing methods in this class, so we should probably run with it for now and refactor at a later date, if we get the desire.

Reply:
I originally thought about an extension method - but then there's no access to the services collection - so we'd need to sort that out to support extension methods.